dispatcher/stateful.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+module.exports = function StatefulDispatcher(dispatcher, data, onData) {
```

Comment:
yes, I figured that.

My impression was, that `BrowserDispatcher` and `ServerDispatcher` are awful names.

So, the `StatefulDispatcher` wraps state and retains the root data tree over time, notifying its client about each data change via the callback.

The `StaticDispatcher` on the other hand does not wrap or retain data across dispatches, and will call the callback only once, as soon as no further actions are pending. I am not so convinced about that name, though. Better suggestions?